ykman/driver_otp.py

```diff
@@ -123,12 +126,16 @@ def _read_pid(self):
         check(ykpers.yk_get_key_vid_pid(self._dev, byref(vid), byref(pid)))
         return PID(pid.value)
 
-    def _read_serial(self):
+    def _read_serial(self, attempts=3):
         serial = c_uint()
-        if ykpers.yk_get_serial(self._dev, 0, 0, byref(serial)):
-            return serial.value
-        else:
-            return None  # Serial not visible
+        for i in range(1, attempts):
+            if ykpers.yk_get_serial(self._dev, 0, 0, byref(serial)):
+                return serial.value
+            else:
+                logger.debug(
+                    'Failed to read serial from device. Attempt {}.'.format(i))
```

Comment:
The logger methods have formatting built in - see review summary.

Reply:
Will update the logging, thx.